packages/tupaia-web/src/features/MapOverlays/PolygonLayer.tsx

```diff
@@ -23,22 +24,30 @@ const ChildEntities = ({ entities }: { entities: EntityResponse['children'] }) =
   );
 };
 
-const SiblingEntities = ({ entity }: { entity: EntityResponse }) => {
+const SiblingEntities = ({
+  parentEntityCode,
+  activeEntityCode,
+}: {
+  parentEntityCode: Entity['code'];
+  activeEntityCode: Entity['code'];
+}) => {
   const { projectCode } = useParams();
   const { data: siblingEntities, isLoading } = useEntitiesWithLocation(
     projectCode,
-    entity.parentCode,
+    parentEntityCode,
   );
 
-  if (isLoading || !siblingEntities) {
+  if (isLoading || !siblingEntities || siblingEntities.length === 0) {
     return null;
   }
 
-  const children = siblingEntities?.children?.filter(e => e.code !== entity.code) || [];
+  const children = siblingEntities.filter(
+    (entity: EntityResponse) => entity.code !== activeEntityCode,
```

Comment:
I wonder if the types need to be added here, or if having them in the `useEntitiesWithLocation` would solve any TS errors with that? Might save us writing it out each time

Reply:
oh yes you are correct. I didn't need them here. I must have added them in the wrong order. thanks